index.js

```diff
@@ -696,11 +726,20 @@ answer.encode = function (a, buf, offset) {
 
   buf.writeUInt16BE(types.toType(a.type), offset)
 
-  let klass = classes.toClass(a.class === undefined ? 'IN' : a.class)
-  if (a.flush) klass |= FLUSH_MASK // the 1st bit of the class is the flush bit
-  buf.writeUInt16BE(klass, offset + 2)
-
-  buf.writeUInt32BE(a.ttl || 0, offset + 4)
+  if (a.type === 'OPT') {
+    if (a.name !== '.') {
+      throw new Error('OPT name must be root.')
+    }
```

Comment:
At this point, we already have encoded the name, I think this check should be done before that if we want to keep it. I'd also be okay with just setting it to '.' if undefined.

Reply:
Alternately, just assume the name is "." so it doesn't need to be specified.

Reply:
Yes, that could work to, assume `.` when its undefined, otherwise check the value.